lib/cdo/log_object.rb

```diff
@@ -0,0 +1,85 @@
+# LogObject is a simple container to save errors and important info
+# when executing a process. It can also time block execution and log the result.
+#
+# LogObject is helpful when we want to
+# - Prevent non-fatal errors from stopping process execution but still want to know about them.
+# - Bubble up combined errors and info from lower levels to higher levels.
+#   Even to external components such as HoneyBadger or Slack channel.
+
+class LogObject
```

Comment:
Is it redundant to include Object in the class name?

Reply:
Maybe rename it to `LogCollection`?

Reply:
How about `LogCollector` to emphasize that it is an object actively measure and collect info.